frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel.tsx

```diff
@@ -38,48 +68,119 @@ export const NavigationPanel = () => {
 
         <div className={'px-2 flex flex-col'}>
           {folders.map((folder, index) => (
-            <div key={index}>
-              <div className={'px-4 py-2 my-2 flex items-center justify-between rounded-lg hover:bg-surface-2'}>
-                <div
-                  onClick={() => setFolderOpen(folder.id, !isFolderOpen[folder.id])}
-                  className={'flex items-center flex-1 min-w-0 cursor-pointer '}
-                >
+            <div key={index} className={'my-2'}>
+              <div
+                onClick={() => renamingFolderId !== folder.id && setFolderOpen(folder.id, !isFolderOpen[folder.id])}
+                className={
+                  'px-4 py-2 cursor-pointer flex items-center justify-between rounded-lg hover:bg-surface-2 ' +
+                  (renamingFolderId === folder.id ? 'bg-main-warning hover:bg-main-warning' : '')
+                }
+              >
+                <div className={'flex items-center flex-1 min-w-0'}>
                   <div
                     className={`mr-2 transition-transform duration-500 ${isFolderOpen[folder.id] ? 'rotate-180' : ''}`}
                   >
                     <img className={''} src={'/images/home/drop_down_show.svg'} alt={''} />
                   </div>
-                  <span className={'whitespace-normal min-w-0 flex-1'}>{folder.title}</span>
-                  {/* <input
-                    className={'whitespace-normal min-w-0 flex-1'}
-                    value={folder.title}
-                    onChange={(e) => onFolderChange(folder.id, e.target.value)}
-                  />*/}
+                  {renamingFolderId === folder.id ? (
+                    <input
+                      className={'whitespace-normal min-w-0 flex-1 '}
+                      value={folder.title}
+                      onKeyDown={(e) => e.code === 'Enter' && completeFolderRename()}
+                      onChange={(e) => onFolderChange(folder.id, e.target.value)}
+                    />
+                  ) : (
+                    <span className={'whitespace-nowrap overflow-ellipsis overflow-hidden min-w-0 flex-1'}>
+                      {folder.title}
+                    </span>
+                  )}
+                </div>
+                <div className={'flex items-center relative'}>
+                  <button
```

Comment:
This component contains too many buttons. Is there a way to optimize? 
For example, this button seems like it can be an independent component.

Reply:
makes sense, I refactored the whole Navigation